File: src/types/menu.ts
export type Locale = "tr" | "en";

/** Every customer-visible string carries both languages together. */
export type Localized = Record<Locale, string>;

export type DietaryTag =
  | "vegan"
  | "vegetarian"
  | "spicy"
  | "chefs-choice"
  | "caffeine"
  | "contains-alcohol-free-mixer";

export type ImageSource = "house" | "stock" | "none";

export interface ImageRef {
  /** `house` is MiST's own photography; `stock` is licensed placeholder. */
  source: ImageSource;
  src: string;
  alt: Localized;
  /** Tiny base64 LQIP so cards do not pop in. */
  blurDataURL?: string;
}

export interface ModifierOption {
  slug: string;
  name: Localized;
  /** Signed delta in kuruş against the parent item's price. */
  priceDeltaMinor: number;
  isDefault: boolean;
  isAvailable: boolean;
  sortOrder: number;
}

export interface ModifierGroup {
  slug: string;
  name: Localized;
  selection: "single" | "multiple";
  isRequired: boolean;
  minSelect: number;
  maxSelect: number;
  sortOrder: number;
  options: ModifierOption[];
}

export interface MenuItem {
  slug: string;
  categorySlug: string;
  name: Localized;
  description: Localized;
  /**
   * Price in kuruş (1/100 TRY) as an integer. Never a float: split-bill
   * arithmetic on floats is how rounding bugs reach a customer's receipt.
   */
  priceMinor: number;
  image: ImageRef;
  tags: DietaryTag[];
  /**
   * Absent from the source data and never inferred. Rendering a guessed
   * allergen list for a real restaurant is a food-safety claim, not a UI
   * detail — these stay empty until the cafe supplies them.
   */
  allergens: string[];
  calories?: number;
  modifierGroups: ModifierGroup[];
  isAvailable: boolean;
  sortOrder: number;
  /** Pre-folded haystack so search does not re-normalise on every keystroke. */
  searchBlob: Localized;
}

export interface MenuCategory {
  slug: string;
  groupSlug: string;
  name: Localized;
  /** Moved out of a Turkish-keyed lookup in the component and into the data. */
  icon: string;
  image?: ImageRef;
  sortOrder: number;
  items: MenuItem[];
}

export interface MenuGroup {
  slug: string;
  name: Localized;
  icon: string;
  sortOrder: number;
}

export interface MenuDocument {
  version: string;
  generatedAt: string;
  currency: "TRY";
  /** Explicit so the UI can state it rather than implying completeness. */
  allergenDataAvailable: boolean;
  calorieDataAvailable: boolean;
  groups: MenuGroup[];
  categories: MenuCategory[];
}

File: backend/src/Mist.Domain/Entities/Menu.cs
using Mist.Domain.Enums;
using Mist.Domain.ValueObjects;

namespace Mist.Domain.Entities;

public sealed class MenuGroup
{
    public Guid Id { get; set; } = Guid.CreateVersion7();
    public string Slug { get; set; } = string.Empty;
    public LocalizedText Name { get; set; } = LocalizedText.Empty;
    public string Icon { get; set; } = string.Empty;
    public int SortOrder { get; set; }

    public ICollection<Category> Categories { get; set; } = [];
}

public sealed class Category
{
    public Guid Id { get; set; } = Guid.CreateVersion7();
    public string Slug { get; set; } = string.Empty;
    public Guid MenuGroupId { get; set; }
    public MenuGroup? MenuGroup { get; set; }
    public LocalizedText Name { get; set; } = LocalizedText.Empty;
    public string Icon { get; set; } = string.Empty;
    public string? ImageUrl { get; set; }
    public int SortOrder { get; set; }
    public bool IsActive { get; set; } = true;

    public ICollection<MenuItem> Items { get; set; } = [];
}

public sealed class MenuItem
{
    public Guid Id { get; set; } = Guid.CreateVersion7();
    public string Slug { get; set; } = string.Empty;
    public Guid CategoryId { get; set; }
    public Category? Category { get; set; }

    public LocalizedText Name { get; set; } = LocalizedText.Empty;
    public LocalizedText Description { get; set; } = LocalizedText.Empty;

    /// <summary>Price in kurus. Integer throughout — a split bill must not
    /// accumulate float rounding error on a customer's receipt.</summary>
    public int PriceMinor { get; set; }

    public string? ImageUrl { get; set; }
    public LocalizedText ImageAlt { get; set; } = LocalizedText.Empty;

    public string[] Tags { get; set; } = [];

    /// <summary>Empty until the cafe supplies real data. Never inferred:
    /// an allergen list is a food-safety claim, not a UI detail.</summary>
    public string[] Allergens { get; set; } = [];

    public int? Calories { get; set; }
    public bool IsAvailable { get; set; } = true;
    public int SortOrder { get; set; }

    public ICollection<ModifierGroup> ModifierGroups { get; set; } = [];
}

public sealed class ModifierGroup
{
    public Guid Id { get; set; } = Guid.CreateVersion7();
    public string Slug { get; set; } = string.Empty;
    public Guid MenuItemId { get; set; }
    public MenuItem? MenuItem { get; set; }
    public LocalizedText Name { get; set; } = LocalizedText.Empty;
    public ModifierSelection Selection { get; set; } = ModifierSelection.Single;
    public bool IsRequired { get; set; }
    public int MinSelect { get; set; }
    public int MaxSelect { get; set; } = 1;
    public int SortOrder { get; set; }

    public ICollection<ModifierOption> Options { get; set; } = [];
}

public sealed class ModifierOption
{
    public Guid Id { get; set; } = Guid.CreateVersion7();
    public string Slug { get; set; } = string.Empty;
    public Guid ModifierGroupId { get; set; }
    public ModifierGroup? ModifierGroup { get; set; }
    public LocalizedText Name { get; set; } = LocalizedText.Empty;
    public int PriceDeltaMinor { get; set; }
    public bool IsDefault { get; set; }
    public bool IsAvailable { get; set; } = true;
    public int SortOrder { get; set; }
}
